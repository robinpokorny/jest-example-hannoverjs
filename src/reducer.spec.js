import reducer from './reducer'

test('creates initial state', () => {
  expect(reducer(undefined, {})).toMatchSnapshot()
})

test('changes name when logged in', () => {
  const action = {
    type: 'CHANGE_NAME',
    payload: 'Gregor Samsa'
  }

  expect(reducer(undefined, action)).toMatchSnapshot()
})

test('does not change name when not logged in', () => {
  const state = {
    loggedIn: false
  }
  const action = {
    type: 'CHANGE_NAME',
    payload: 'Gregor Samsa'
  }

  expect(reducer(state, action)).toMatchSnapshot()
})
