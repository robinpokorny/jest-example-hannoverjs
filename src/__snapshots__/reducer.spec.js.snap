// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes name when logged in 1`] = `
Object {
  "loggedIn": true,
  "name": "Gregor Samsa",
}
`;

exports[`creates initial state 1`] = `
Object {
  "loggedIn": true,
  "name": null,
}
`;

exports[`does not change name when not logged in 1`] = `
Object {
  "loggedIn": false,
  "name": null,
}
`;
